extensions/amp-story-audio-sticker/0.1/amp-story-audio-sticker.js

```diff
@@ -33,17 +110,118 @@ export class AmpStoryAudioSticker extends AMP.BaseElement {
         </div>
       </div>
     );
+
+    const pageEl = closestAncestorElementBySelector(
+      this.element,
+      'amp-story-page'
+    );
+    this.pageId_ = pageEl.getAttribute('id');
+
+    Services.storyStoreServiceForOrNull(this.win).then((storeService) => {
+      this.storeService_ = storeService;
+      this.storeService_.subscribe(StateProperty.MUTED_STATE, (muted) => {
+        this.onMutedStateChange_(muted);
+      });
+      this.initializeListeners_();
+    });
   }
 
-  /** @override */
-  layoutCallback() {
-    return Services.storyStoreServiceForOrNull(this.win).then(
-      (storeService) => {
-        this.element.addEventListener('click', () =>
-          storeService.dispatch(Action.TOGGLE_MUTED, false)
+  /** @private */
+  maybeInitializeDefaultSticker_() {
+    if (!this.useDefaultSticker_) {
+      return;
+    }
+    this.sticker_ =
+      this.element.getAttribute('sticker') || FALLBACK_DEFAULT_STICKER;
+
+    // Remove all existing child elements and add only pretap and posttap elements.
+    this.element.replaceChildren();
+    this.element.appendChild(
+      <amp-story-audio-sticker-pretap>
+        <amp-img
+          width={DEFAULT_STICKERS[this.sticker_].width}
+          height={DEFAULT_STICKERS[this.sticker_].height}
+          layout="responsive"
+        ></amp-img>
+      </amp-story-audio-sticker-pretap>
+    );
+    this.element.appendChild(
+      <amp-story-audio-sticker-posttap>
+        <amp-img
+          width={DEFAULT_STICKERS[this.sticker_].width}
+          height={DEFAULT_STICKERS[this.sticker_].height}
+          layout="responsive"
+        ></amp-img>
+      </amp-story-audio-sticker-posttap>
+    );
+  }
+
+  /**
+   * Hide the sticker after the story gets unmuted:
+   * 1) if the sticker is on the active page, hide it after showing posttap sticker for 4 seconds.
+   * 2) if the sticker is not on the active page, hide it immediately since it is in the background.
+   * @param {boolean} muted
+   * @private
+   */
+  onMutedStateChange_(muted) {
+    if (!muted) {
+      if (
+        this.pageId_ === this.storeService_.get(StateProperty.CURRENT_PAGE_ID)
+      ) {
+        this.hideStickerTimeout_ = setTimeout(
+          () => this.toggleStickerState_('hide', true),
+          HIDE_STICKER_DELAY_DURATION
         );
+      } else {
+        this.toggleStickerState_('hide', true);
       }
+    } else {
+      this.toggleStickerState_('hide', false);
+      this.hideStickerTimeout_ && clearTimeout(this.hideStickerTimeout_);
+      this.hideStickerTimeout_ = null;
```

Comment:
You can do all this with CSS and `transition-delay: 4s` when it's muted. No need to handle JS timeouts.

Reply:
Thanks for the advice! Done!